tests/integration/update_cluster/irsa/kubernetes.tf

```diff
@@ -1,25 +1,27 @@
-  node_autoscaling_group_ids            = [aws_autoscaling_group.nodes-minimal-example-com.id]
-  node_security_group_ids               = [aws_security_group.nodes-minimal-example-com.id]
-  node_subnet_ids                       = [aws_subnet.us-test-1a-minimal-example-com.id]
-  nodes_role_arn                        = aws_iam_role.nodes-minimal-example-com.arn
-  nodes_role_name                       = aws_iam_role.nodes-minimal-example-com.name
-  region                                = "us-test-1"
-  route_table_public_id                 = aws_route_table.minimal-example-com.id
-  subnet_us-test-1a_id                  = aws_subnet.us-test-1a-minimal-example-com.id
-  vpc_cidr_block                        = aws_vpc.minimal-example-com.cidr_block
-  vpc_id                                = aws_vpc.minimal-example-com.id
+  cluster_name                                      = "minimal.example.com"
+  default-myserviceaccount_role_arn                 = aws_iam_role.myserviceaccount-default-sa-minimal-example-com.arn
+  default-myserviceaccount_role_name                = aws_iam_role.myserviceaccount-default-sa-minimal-example-com.name
+  iam_openid_connect_provider_arn                   = aws_iam_openid_connect_provider.minimal-example-com.arn
+  iam_openid_connect_provider_issuer                = "discovery.example.com/minimal.example.com"
+  master_autoscaling_group_ids                      = [aws_autoscaling_group.master-us-test-1a-masters-minimal-example-com.id]
+  master_security_group_ids                         = [aws_security_group.masters-minimal-example-com.id]
+  masters_role_arn                                  = aws_iam_role.masters-minimal-example-com.arn
+  masters_role_name                                 = aws_iam_role.masters-minimal-example-com.name
+  myapp-myotherserviceaccount_role_arn              = aws_iam_role.myotherserviceaccount-myapp-sa-minimal-example-com.arn
+  myapp-myotherserviceaccount_role_name             = aws_iam_role.myotherserviceaccount-myapp-sa-minimal-example-com.name
+  node_autoscaling_group_ids                        = [aws_autoscaling_group.nodes-minimal-example-com.id]
+  node_security_group_ids                           = [aws_security_group.nodes-minimal-example-com.id]
+  node_subnet_ids                                   = [aws_subnet.us-test-1a-minimal-example-com.id]
+  nodes_role_arn                                    = aws_iam_role.nodes-minimal-example-com.arn
+  nodes_role_name                                   = aws_iam_role.nodes-minimal-example-com.name
+  region                                            = "us-test-1"
+  route_table_public_id                             = aws_route_table.minimal-example-com.id
+  subnet_us-test-1a_id                              = aws_subnet.us-test-1a-minimal-example-com.id
+  test- * -myserviceaccountwithwildcardns_role_arn  = aws_iam_role.myserviceaccountwithwildcardns-test-wildcard-sa-minimal-e-im2tl2.arn
```

Comment:
There is an `ExportWithID` field of the IAMRole task that is only used in building these `local` and `output` names:

https://github.com/kubernetes/kops/blob/6612ba12a596f46c42bc80938ceac05e067a5574/upup/pkg/fi/cloudup/awstasks/iamrole.go#L352-L355

The model sets it in the task here:

https://github.com/kubernetes/kops/blob/6612ba12a596f46c42bc80938ceac05e067a5574/pkg/model/awsmodel/iam.go#L207-L213

We have two options:
* Don't set ExportWithID when the SA namespace contains `*`
* Do the same strings.ReplaceAll that we do with `name` on ExportWithID as well.

Reply:
Thanks @rifelpet 👍